tests/test_dirty_local.py

```diff
@@ -0,0 +1,117 @@
+import os
+from pathlib import Path
+from shutil import copy2, copytree
+
+import pytest
+from plumbum import local
+from plumbum.cmd import git
+
+import copier
+from copier.errors import DirtyLocalWarning
+from copier.main import run_copy, run_update
+
+from .helpers import DATA, PROJECT_TEMPLATE, build_file_tree, filecmp
+
+
+def test_copy(tmp_path_factory):
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
+
+    # dirs_exist_ok not available in Python 3.7
+    for item in os.listdir(PROJECT_TEMPLATE):
+        item_src_path = os.path.join(PROJECT_TEMPLATE, item)
+        item_dst_path = os.path.join(src, item)
+        if os.path.isdir(item_src_path):
+            copytree(item_src_path, item_dst_path)
+        else:
+            copy2(item_src_path, item_dst_path)
+
+    with local.cwd(src):
+        git("init")
+
+    with pytest.warns(DirtyLocalWarning):
+        copier.copy(str(src), str(dst), data=DATA, quiet=True)
+
+    generated = (dst / "pyproject.toml").read_text()
+    control = (Path(__file__).parent / "reference_files" / "pyproject.toml").read_text()
+    assert generated == control
+
+    # assert template still dirty
+    with local.cwd(src):
+        assert bool(git("status", "--porcelain").strip())
+
+
+# Will fail due to lingering deleted file.
+# HACK https://github.com/copier-org/copier/issues/461
+# TODO Remove xfail decorator when fixed.
+@pytest.mark.xfail(strict=True)
+def test_update(tmp_path_factory):
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
+
+    build_file_tree(
+        {
+            src
+            / ".copier-answers.yml.jinja": """\
+                # Changes here will be overwritten by Copier
+                {{ _copier_answers|to_nice_yaml }}
+            """,
+            src
+            / "copier.yml": """\
+                _envops:
+                    "keep_trailing_newline": True
+            """,
+            src
+            / "aaaa.txt": """
+                Lorem ipsum
+            """,
+            src
+            / "to_delete.txt": """
```

Comment:
Since the failure from this test is in this file, then please remove this part of the test and remove the xfail decorator. If the algorithm is fine, the test will pass. Later, when #461 is fixed, it will include the needed tests.

Reply:
kept file, but removed the assert. added a node referencing the linked issue